include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -407,93 +409,107 @@ class area_formulas
     /*
         Compute the ellipsoidal correction of a geodesic (or shperical) segment
     */
-    template <
-                template <typename, bool, bool, bool, bool, bool> class Inverse,
-                typename PointOfSegment,
-                typename SpheroidConst
-             >
-    static inline return_type_ellipsoidal ellipsoidal(PointOfSegment const& p1,
-                                                      PointOfSegment const& p2,
-                                                      SpheroidConst const& spheroid_const)
+    template
+    <
+        typename FormulaPolicy,
+        typename PointOfSegment,
+        typename SpheroidConst
+    >
+    static inline auto ellipsoidal(PointOfSegment const& p1,
+                                   PointOfSegment const& p2,
+                                   SpheroidConst const& spheroid_const)
     {
         return_type_ellipsoidal result;
 
-        // Azimuth Approximation
+        CT const lon1r = get_as_radian<0>(p1);
+        CT const lat1r = get_as_radian<1>(p1);
+        CT const lon2r = get_as_radian<0>(p2);
+        CT const lat2r = get_as_radian<1>(p2);
 
-        typedef Inverse<CT, false, true, true, false, false> inverse_type;
-        typedef typename inverse_type::result_type inverse_result;
+        // Azimuth Approximation
 
-        inverse_result i_res = inverse_type::apply(get_as_radian<0>(p1),
-                                                   get_as_radian<1>(p1),
-                                                   get_as_radian<0>(p2),
-                                                   get_as_radian<1>(p2),
-                                                   spheroid_const.m_spheroid);
+        auto const i_res = FormulaPolicy::template inverse
```

Comment:
FormulaPolicy is strategies-related. It was implemented in order to allow t pick formula in strategies. They are implemented in strategies directory. I think bare formula should be used here.

Reply:
I am aware that `FormulaPolicy` was implemented to pick formulas in strategies and we are in formulas here, but I found it handy since base formula cannot be passed here since karney inverse formula has a different templated type i.e. 
`inverse<typename CT, bool, bool, bool, bool, bool, size_t >` while the other inverse formulas have `inverse<typename CT, bool, bool, bool, bool, bool>`

Reply:
If you instantiate it here using FormulaPolicy the default `SeriesOrder = 8` of karney's formula is used anyway. The interface of formulas and policies are the same, you still have to pass the same parameters because other formulas does no take `SeriesOrder`.

So this is not about passing `SeriesOrder` but only about the template parameter.

So why not simply have a `detail` version allowing to pass `SeriesOrder` for testing purposes and then the "official" version implementing the same interface as all other formulas?

Btw, there is also the `SeriesOrder` passed into the strategy by the user. Are they corresponding? If not shouldn't we simply remove them and always use our defaults?

Reply:
> So why not simply have a detail version allowing to pass SeriesOrder for testing purposes and then the "official" version implementing the same interface as all other formulas?

This is a good idea. I implemented a `detail` version of `karney_inverse`. 

I kept the interface of area strategy the same. `SeriesOrder` in area strategy are unrelated to series order in inverse formulas. I added some examples in area tests to show this.